### ISSUE TITLE
Flux.bufferTimeout : Does this method produce duplicate data to subscriber ?

### ISSUE THREAD

**Issue by yuncheolkim** [reactor/reactor-core#1532]
I did a test. and subscriber received  duplicate data. Is this normal behavior?
```
@Test
    public void test1() throws InterruptedException {
        HashSet<Long> hashSet = new HashSet<>();
        Flux.interval(Duration.ofMillis(2)).bufferTimeout(2, Duration.ofMillis(2)).subscribe(
                values -> {
                    System.out.println(values);
                    values.forEach(aLong -> {
                        Assert.assertTrue("error:"+aLong,hashSet.add(aLong));
                    });
                }
        );
        Thread.sleep(10000);

    }
```
output
```
[1951, 1952]
[1953, 1954]
[1955, 1956]
[1955, 1956]
16:47:46.064 [parallel-1] ERROR reactor.core.scheduler.Schedulers - Scheduler worker in group main failed with an uncaught exception
reactor.core.Exceptions$ErrorCallbackNotImplemented: java.lang.AssertionError: error:1955
Caused by: java.lang.AssertionError: error:1955
```
There has two same array

**Comment by bsideup** [comment]
Hi @yuncheolkim,

I tried to run you test with the latest Reactor version and II did not observe the error you have.
Which Reactor version are you using?

**Comment by yuncheolkim** [comment]
My Reactor version is `reactor-core-3.2.5.RELEASE` .
This is not 100% happen.

**Comment by bsideup** [comment]
Okay, after running the test for some minutes I got the exception. Thanks, we will investigate!

**Comment by bsideup** [comment]
@yuncheolkim thanks for reporting it! 
There was a race condition and we fixed it in both 3.1.x and 3.2.x.
